packages/block-editor/src/components/use-block-display-information/index.js

```diff
@@ -22,49 +25,67 @@ import { store as blockEditorStore } from '../../store';
  */
 
 /**
- * Hook used to try to find a matching block variation and return
- * the appropriate information for display reasons. In order to
- * to try to find a match we need to things:
+ * Hook used appropriate information for display reasons.
+ * It takes variations and custom `label` function into account.
+ *
+ * In order to try to find a variation match we need to things:
  * 1. Block's client id to extract it's current attributes.
  * 2. A block variation should have set `isActive` prop to a proper function.
  *
- * If for any reason a block variaton match cannot be found,
- * the returned information come from the Block Type.
+ * Value from custom `label` function is prioritized, if that's not found
+ * then the block variation' information is returned. If both are missing
+ * then the information is returned from the Block Type.
  * If no blockType is found with the provided clientId, returns null.
  *
  * @param {string} clientId Block's client id.
  * @return {?WPBlockDisplayInformation} Block's display information, or `null` when the block or its type not found.
  */
-
 export default function useBlockDisplayInformation( clientId ) {
 	return useSelect(
 		( select ) => {
 			if ( ! clientId ) return null;
+
 			const { getBlockName, getBlockAttributes } = select(
 				blockEditorStore
 			);
 			const { getBlockType, getBlockVariations } = select( blocksStore );
+
 			const blockName = getBlockName( clientId );
 			const blockType = getBlockType( blockName );
 			if ( ! blockType ) return null;
+
 			const variations = getBlockVariations( blockName );
-			const blockTypeInfo = {
-				title: blockType.title,
-				icon: blockType.icon,
-				description: blockType.description,
-			};
-			if ( ! variations?.length ) return blockTypeInfo;
 			const attributes = getBlockAttributes( clientId );
-			const match = variations.find( ( variation ) =>
-				variation.isActive?.( attributes, variation.attributes )
-			);
-			if ( ! match ) return blockTypeInfo;
+
+			const dynamicTitle = getBlockLabel( blockType, attributes );
+			const variationInfo = getVariationInfo( variations, attributes );
+
+			const title =
```

Comment:
The use of the `getBlockLabel` to generate a "dynamic" title makes sense to me, but, with that being said, I don't have a lot of historical context with `use-block-display-information`.

Also, @david-szabo97, do you happen to know what the exact differences are between a block label and title? The label appears to be a canonical name, while the title is something customized by the user (definitely correct me if I'm wrong).

Reply:
> The label appears to be a canonical name, while the title is something customized by the user (definitely correct me if I'm wrong).

I don't think the `title` is really a thing that can be defined anywhere, it seems to be what we use as a variable name when deciding whether to show the block's name (e.g. 'Heading') or the label (the value returned by the `__experimentalLabel` property on a block).

Reply:
Oh I see -- I had it reversed in my head. This makes sense. Thanks @talldan!